pkg/kubelet/cm/devicemanager/manager_test.go

```diff
@@ -1347,6 +1347,118 @@ func TestInitContainerDeviceAllocation(t *testing.T) {
+	}
+	tmpDir, err := os.MkdirTemp("", "checkpoint")
+	as.Nil(err)
+	defer os.RemoveAll(tmpDir)
+
+	testManager, err := getTestManager(tmpDir, podsStub.getActivePods, testResources)
+	as.Nil(err)
+
+	containerRestartPolicyAlways := v1.ContainerRestartPolicyAlways
+	podWithPluginResourcesInRestartableInitContainers := &v1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			UID: uuid.NewUUID(),
+		},
+		Spec: v1.PodSpec{
+			InitContainers: []v1.Container{
+				{
+					Name: string(uuid.NewUUID()),
+					Resources: v1.ResourceRequirements{
+						Limits: v1.ResourceList{
+							v1.ResourceName(res1.resourceName): *oneDevice,
+						},
+					},
+					RestartPolicy: &containerRestartPolicyAlways,
+				},
+				{
+					Name: string(uuid.NewUUID()),
+					Resources: v1.ResourceRequirements{
+						Limits: v1.ResourceList{
+							v1.ResourceName(res1.resourceName): *twoDevice,
+						},
+					},
+					RestartPolicy: &containerRestartPolicyAlways,
+				},
+			},
+			Containers: []v1.Container{
+				{
+					Name: string(uuid.NewUUID()),
+					Resources: v1.ResourceRequirements{
+						Limits: v1.ResourceList{
+							v1.ResourceName(res1.resourceName): *oneDevice,
+						},
+					},
+				},
+				{
+					Name: string(uuid.NewUUID()),
+					Resources: v1.ResourceRequirements{
+						Limits: v1.ResourceList{
+							v1.ResourceName(res1.resourceName): *twoDevice,
+						},
+					},
+				},
+			},
+		},
+	}
+	podsStub.updateActivePods([]*v1.Pod{podWithPluginResourcesInRestartableInitContainers})
+	for _, container := range podWithPluginResourcesInRestartableInitContainers.Spec.InitContainers {
+		err = testManager.Allocate(podWithPluginResourcesInRestartableInitContainers, &container)
+	}
+	for _, container := range podWithPluginResourcesInRestartableInitContainers.Spec.Containers {
+		err = testManager.Allocate(podWithPluginResourcesInRestartableInitContainers, &container)
+	}
+	as.Nil(err)
+	podUID := string(podWithPluginResourcesInRestartableInitContainers.UID)
+	restartableInitCont1 := podWithPluginResourcesInRestartableInitContainers.Spec.InitContainers[0].Name
+	restartableInitCont2 := podWithPluginResourcesInRestartableInitContainers.Spec.InitContainers[1].Name
+	normalCont1 := podWithPluginResourcesInRestartableInitContainers.Spec.Containers[0].Name
+	normalCont2 := podWithPluginResourcesInRestartableInitContainers.Spec.Containers[1].Name
+	restartableInitCont1Devices := testManager.podDevices.containerDevices(podUID, restartableInitCont1, res1.resourceName)
+	restartableInitCont2Devices := testManager.podDevices.containerDevices(podUID, restartableInitCont2, res1.resourceName)
+	normalCont1Devices := testManager.podDevices.containerDevices(podUID, normalCont1, res1.resourceName)
+	normalCont2Devices := testManager.podDevices.containerDevices(podUID, normalCont2, res1.resourceName)
+	as.Equal(1, restartableInitCont1Devices.Len())
+	as.Equal(2, restartableInitCont2Devices.Len())
+	as.Equal(1, normalCont1Devices.Len())
+	as.Equal(2, normalCont2Devices.Len())
```

Comment:
Sorry for being pedantic, this is mostly a NIT:
I can see that you are following the conventions in this file:`pkg/kubelet/cm/devicemanager/manager_test.go` but in the rest of the project, the convention is to place the obtained value before the expected value. Maybe we can fix that in this PR and comply with the project conventions?

Reply:
I understand your concern but it seems inappropriate for this function.
`as.Equal` function is supposed to specify the expected value followed by the actual value, and it affects the failure message.

```
$ go doc github.com/stretchr/testify/require.Equal
package require // import "github.com/stretchr/testify/require"

func Equal(t TestingT, expected interface{}, actual interface{}, msgAndArgs ...interface{})
    Equal asserts that two objects are equal.

        assert.Equal(t, 123, 123)

    Pointer variable equality is determined based on the equality of the
    referenced values (as opposed to the memory addresses). Function equality
    cannot be determined and will always fail.
```

```
--- FAIL: TestRestartableInitContainerDeviceAllocation (0.05s)
    manager_test.go:1451: 
                Error Trace:    /home/gjkim/go/src/k8s.io/kubernetes/pkg/kubelet/cm/devicemanager/manager_test.go:1451
                Error:          Not equal: 
                                expected: 3
                                actual  : 2
                Test:           TestRestartableInitContainerDeviceAllocation
```

Reply:
Ack, Thanks for double checking this.